src/libasr/ASR.asdl

```diff
@@ -103,12 +103,12 @@ symbol
         ttype type, symbol? parent)
     | UnionType(symbol_table symtab, identifier name, identifier* dependencies,
         identifier* members, abi abi, access access, symbol? parent)
-    | Variable(symbol_table parent_symtab, identifier name, intent intent,
-        expr? symbolic_value, expr? value, storage_type storage, ttype type,
+    | Variable(symbol_table parent_symtab, identifier name, identifier* dependencies,
+        intent intent, expr? symbolic_value, expr? value, storage_type storage, ttype type,
```

Comment:
I see that we always use `identifier* dependencies` for all symbols, such as Functions. What is the convention --- does the dependency have to be in the parent scope, i.e. the symbol table that contains the given symbol?

What if a function calls another function from another module? What if a variable declaration depends on another variable from another module? I guess then it would be ExternalSymbol in the same scope.

What if it depends on something that is further up in the scope (parent's parent)?

Is the logic that you can find it by "resolving" it recursively?

Why cannot we just point to the symbols directly?

Reply:
Let's document this in here: https://github.com/lfortran/lfortran/blob/bb49552e0fc51706ba4f0772636451c5b0aab8fc/doc/src/asr/asr_nodes/symbol_nodes/symbol.md, we should probably split it into one file per symbol node, and document what the intention and restrictions are of each argument.

Reply:
I am following the pattern of storing dependencies in `Module`. 

> Why cannot we just point to the symbols directly?

We can. There's nothing preventing us from doing so. Its a judgement call. Store the dependencies as strings and resolve them using `current_scope->resolve_symbol(string_name)` or just use the symbols directly. No strong opinions, both should work. What do you suggest to do?

Reply:
Strings feel easier to manage though. 😅.